custom_components/pioneer_async/pioneer_avr.py

```diff
@@ -860,11 +862,20 @@ async def set_volume_level(self, volume, zone="1"):
             or (zone != "1" and volume > MAX_VOLUME_ZONEX)
         ):
             raise ValueError(f"volume {volume} out of range for zone {zone}")
-        vol_len = 3 if zone == "1" else 2
-        vol_prefix = str(volume).zfill(vol_len)
-        return await self.send_command(
-            "set_volume_level", zone, prefix=vol_prefix, ignore_error=False
-        )
+        if self.volume_steps:
+            current_volume = self.volume.get(zone)
+            steps = abs(round((volume - current_volume) / 2))
```

Comment:
`volume` is measured in the relevant units for the zone (0.5db increments for Zone 1, 1db increments for other zones.) So should this be dividing the volume difference by 2 to determine how many commands to send? On my AVR, the volume up/down commands skip by 0.5db on Zone 1 and 1db on HDZone.

Reply:
My VSX-S510 works differently. The volume can be adjusted from 0 to 151 volume units and the volume up/down commands skip by 2 volume units.

Your function `query_device_info()` works well for me.

Do you think adding an exception is a good idea ?

```
if self.model == "VSX-S510":
        self.volume_steps = True
	self.max_volume["1"] = 151
	self.volume_steps_units = [2]
---

if self.volume_steps:
	current_volume = self.volume.get(zone)
	steps = abs(round((volume - current_volume) / self.volume_steps_units[int(zone)]))
		for x in range(steps):
	if volume > current_volume:
		await self.volume_up()
	elif volume < current_volume:
		await self.volume_down()
```

Or group AVR in families is better ?
Normal Family (VSX-5XX, VSX-8XX, VSX-9XX) and Slim Family (VSX-SXXX) ?

Reply:
I am not a great fan of hard-coded exceptions :) However, for now, let's leave it as a separate option that supports VSX-S510 specifically.

Going forward, I would like `PioneerAVR` to be able to determine appropriate values for currently hard-coded defaults for an AVR (eg. min/max volume, volume up/down step) based on model number, as well as HA being able to query those defaults (to use as defaults in the UI) as well as set them if the user chooses to do so via either the UI or `configuration.yaml`. I'd rather complete the separation of the API currently in progress before embarking on this, so I've created crowbarz/aiopioneer#1 to track that work.